src/assets/wise5/services/notebookService.ts

```diff
@@ -76,17 +78,34 @@ export class NotebookService {
     private ConfigService: ConfigService,
     private ProjectService: ProjectService,
     private StudentAssetService: StudentAssetService,
-    private StudentDataService: StudentDataService
+    private StudentDataService: StudentDataService,
+    private UtilService: UtilService
   ) {
     this.notebookItemAnnotationReceivedSubscription = this.StudentDataService.notebookItemAnnotationReceived$.subscribe(
       (args: any) => {
         this.notebookItemAnnotationReceivedSource.next(args);
       }
     );
+    this.addSnipImageListener();
   }
 
-  ngOnDestroy() {
+  addSnipImageListener(): void {
+    this.snipImageHandler = this.createSnipImageHandler();
+    window.addEventListener('snip-image', this.snipImageHandler);
+  }
+
+  createSnipImageHandler(): any {
+    return (event: CustomEvent) => {
+      const imageObject = this.UtilService.getImageObjectFromImageElement(event.detail.target);
+      if (imageObject != null) {
+        this.addNote(imageObject);
+      }
```

Comment:
Can we ensure that calls to UtilService.getImageObjectFromImageElement() won't return a null object? It looks like the function returns null iff the param is null, so can we remove the null-check and assume that it is always called with a non-null param?

Reply:
Yea, I removed the null check.